### PR TITLE
Bump Android SDK to 3.0.6, match kotlin and gradle plugin versions to…

### PR DESCRIPTION
**Pull request by SurvicatePawelGodlewski**
… match those from SDK, bump targetSdk and compileSdk versions to 34, added Proguard rule for obfuscated xml parsers

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -26,7 +26,9 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdkVersion 31
+    namespace 'com.survicate.flutter'
```
Comment (dariusz-survicate) [inline android/build.gradle:29]: We can remove `package="com.survicate.flutter"` from AndroidManifest

Comment (dariusz-survicate) [reply android/build.gradle:29]: or leave it in manifest and here wrap with `if (project.android.hasProperty("namespace"))`

Comment (SurvicatePawelGodlewski) [reply android/build.gradle:29]: I removed package from Manifest

**Comment by SurvicatePawelGodlewski** [comment]
Final changes: 
- bump AGP version to 7.4.2 
- bump Kotlin to 1.8.10
- bump Android Survicate SDK to 3.0.6 (with exclusion of coil 2.5.0)
- added namespace property in build.gradle 
- removed package from manifest
- added proguard rules for xml parsers

**Review by dariusz-survicate — APPROVED** [review]

**Pull request closed without merging**